File: scripts/gen-ground-vehicles.py
"""Generate 5 ground vehicle GLBs via the kiln API."""
import json, urllib.request, subprocess, os

API = "http://localhost:3000/api/kiln/generate"
OUT_DIR = "war-assets/vehicles/ground"
EXPORT_SCRIPT = "scripts/export-glb.ts"

vehicles = [
    {
        "slug": "m151-jeep",
        "prompt": (
            "M151 MUTT jeep - Vietnam War US utility vehicle. Budget: 2000 tris.\n\n"
            "Coordinate system: front toward +Z, Y is up, wheels on ground at Y=0. All parts connected.\n\n"
            "BODY: boxGeo(1.5, 0.6, 3.0) olive 0x556B2F at [0, 0.65, 0]. Main body shell.\n"
            "HOOD: boxGeo(1.3, 0.1, 1.2) olive 0x556B2F at [0, 0.95, 1.2] rotation [-5,0,0]. Angled hood.\n"
            "GRILLE: boxGeo(1.1, 0.4, 0.05) dark 0x333333 at [0, 0.7, 1.8]. Vertical slat front.\n"
            "WINDSHIELD FRAME: boxGeo(1.3, 0.6, 0.04) at [0, 1.2, 0.5]. Frame.\n"
            "WINDSHIELD GLASS: boxGeo(1.2, 0.5, 0.02) glass 0x88aaaa opacity 0.4 at [0, 1.2, 0.5].\n"
            "FRONT FENDERS: Left boxGeo(0.3, 0.15, 1.0) at [-0.75, 0.55, 1.2]. Right mirrored.\n"
            "REAR FENDERS: Left boxGeo(0.25, 0.12, 0.8) at [-0.75, 0.55, -0.8]. Right mirrored.\n"
            "SEATS: Driver boxGeo(0.35, 0.3, 0.35) canvas 0x6B5B3F at [-0.35, 0.85, -0.1]. Passenger at [0.35, 0.85, -0.1].\n"
            "REAR BENCH: boxGeo(1.0, 0.25, 0.3) canvas at [0, 0.8, -0.8].\n"
            "HEADLIGHTS: 2x sphereGeo(0.06, 6, 4) 0xffffcc at [-0.5, 0.8, 1.85] and [0.5, 0.8, 1.85].\n"
            "BUMPER: boxGeo(1.5, 0.1, 0.08) dark 0x333333 at [0, 0.4, 1.85].\n"
            "WHEELS: 4x torusGeo(0.3, 0.12, 8, 12) black 0x222222.\n"
            "- Front left: [−0.8, 0.3, 1.2]. Front right: [0.8, 0.3, 1.2].\n"
            "- Rear left: [−0.8, 0.3, −0.9]. Rear right: [0.8, 0.3, −0.9].\n"
            "WHEEL HUBS: cylinderGeo(0.12, 0.12, 0.1, 6) olive at same positions, rotation [0,0,90].\n"
            "GUN MOUNT RING: torusGeo(0.3, 0.025, 8, 12) dark metal at [0, 1.3, -0.3].\n"
            "SPARE TIRE: torusGeo(0.3, 0.12, 8, 12) at [0, 0.7, -1.6] rotation [90,0,0]."
        ),
    },
    {
        "slug": "m35-truck",
        "prompt": (
            "M35 Deuce-and-a-Half cargo truck - Vietnam War. Budget: 3000 tris.\n\n"
            "Coordinate system: front toward +Z, Y is up, wheels on ground at Y=0. All parts connected.\n\n"
            "CAB: boxGeo(2.2, 1.6, 1.8) olive 0x556B2F at [0, 1.5, 2.0]. Boxy enclosed cab.\n"
            "CAB ROOF: boxGeo(2.2, 0.08, 1.8) at [0, 2.35, 2.0].\n"
            "WINDSHIELD: boxGeo(1.8, 0.8, 0.04) glass 0x88aaaa opacity 0.4 at [0, 1.9, 2.9].\n"
            "SIDE WINDOWS: Left boxGeo(0.04, 0.6, 0.8) glass at [-1.12, 1.9, 2.0]. Right mirrored.\n"
            "GRILLE: boxGeo(1.6, 0.8, 0.06) dark 0x333333 at [0, 1.1, 2.95]. Vertical slat front.\n"
            "HEADLIGHTS: 2x sphereGeo(0.08, 6, 4) 0xffffcc at [-0.6, 1.2, 3.0] and [0.6, 1.2, 3.0].\n"
            "BUMPER: boxGeo(2.2, 0.15, 0.1) dark 0x333333 at [0, 0.55, 3.0].\n"
            "HOOD: boxGeo(2.0, 0.6, 1.2) olive at [0, 1.15, 2.7] rotation [-5,0,0]. Engine hood.\n"
            "CARGO BED FLOOR: boxGeo(2.2, 0.1, 3.8) wood 0x7B5B3A at [0, 0.95, -1.1].\n"
            "BED SIDES: Left boxGeo(0.06, 0.5, 3.8) olive at [-1.1, 1.25, -1.1]. Right mirrored.\n"
            "TAILGATE: boxGeo(2.0, 0.5, 0.06) olive at [0, 1.25, -3.0].\n"
            "CANVAS TOP: Use 4 arched bows + top cover.\n"
            "BOW RIBS: 4x boxGeo(2.0, 0.04, 0.04) dark at [0, 2.1, z] for z = 0.0, -0.9, -1.8, -2.7.\n"
            "CANVAS COVER: boxGeo(2.2, 0.04, 3.8) canvas 0x6B6B4F at [0, 2.15, -1.1]. Top sheet.\n"
            "CANVAS SIDES: Left boxGeo(0.04, 0.9, 3.8) canvas at [-1.12, 1.65, -1.1]. Right mirrored.\n"
            "EXHAUST STACK: cylinderGeo(0.04, 0.04, 1.5, 6) dark 0x333333 at [1.0, 2.0, 1.8].\n"
            "FRONT WHEELS: 2x torusGeo(0.45, 0.15, 8, 12) black 0x222222 at [-1.1, 0.45, 2.2] and [1.1, 0.45, 2.2].\n"
            "REAR WHEELS: 4x (dual axle) torusGeo(0.45, 0.15, 8, 12) at [-1.1, 0.45, -1.0], [1.1, 0.45, -1.0], [-1.1, 0.45, -2.2], [1.1, 0.45, -2.2].\n"
            "WHEEL HUBS: cylinderGeo(0.2, 0.2, 0.12, 6) olive at each wheel, rotation [0,0,90].\n"
            "FRAME RAILS: 2x boxGeo(0.1, 0.15, 6.5) dark 0x333333 at [-0.6, 0.55, 0.0] and [0.6, 0.55, 0.0]."
        ),
    },
    {
        "slug": "m113-apc",
        "prompt": (
            "M113 APC - Vietnam War armored personnel carrier. Budget: 3000 tris.\n\n"
            "Coordinate system: front toward +Z, Y is up, tracks on ground at Y=0. All parts connected.\n\n"
            "HULL BODY: boxGeo(2.6, 1.4, 4.5) olive 0x556B2F at [0, 1.2, 0]. Main aluminum box hull.\n"
            "FRONT GLACIS: boxGeo(2.6, 0.8, 0.06) olive at [0, 1.7, 2.25] rotation [30,0,0]. Angled front plate.\n"
            "REAR RAMP: boxGeo(2.4, 1.2, 0.08) olive at [0, 1.1, -2.28] rotation [-10,0,0]. Drop-down ramp shown closed.\n"
            "TOP DECK: boxGeo(2.6, 0.06, 4.5) olive at [0, 1.92, 0]. Flat top.\n"
            "DRIVER HATCH: boxGeo(0.5, 0.04, 0.5) dark 0x444444 at [-0.6, 1.96, 1.5]. Front left.\n"
            "CUPOLA BASE: cylinderGeo(0.4, 0.4, 0.3, 12) olive at [0.6, 2.2, 1.2]. Commander position.\n"
            "CUPOLA GUN: cylinderGeo(0.025, 0.025, 0.8, 6) gun metal 0x3a3a3a at [0.6, 2.3, 1.7] rotation [90,0,0]. M2 .50 cal.\n"
            "GUN SHIELD: boxGeo(0.5, 0.3, 0.04) gun metal at [0.6, 2.35, 1.55]. Small shield.\n"
            "TRIM VANE: boxGeo(2.4, 0.6, 0.04) olive at [0, 1.6, 2.3] rotation [80,0,0]. Folded flat on glacis.\n"
            "TRACKS: Left track boxGeo(0.3, 0.8, 4.8) dark 0x333333 at [-1.45, 0.4, 0]. Right at [1.45, 0.4, 0].\n"
            "TRACK TOP: Left boxGeo(0.3, 0.06, 4.2) dark at [-1.45, 0.82, 0]. Right mirrored.\n"
            "DRIVE SPROCKET: 2x cylinderGeo(0.35, 0.35, 0.15, 8) dark at [-1.45, 0.5, 2.2] and [1.45, 0.5, 2.2] rotation [0,0,90].\n"
            "IDLER WHEEL: 2x cylinderGeo(0.3, 0.3, 0.12, 8) dark at [-1.45, 0.4, -2.2] and [1.45, 0.4, -2.2] rotation [0,0,90].\n"
            "ROAD WHEELS: 5 per side, cylinderGeo(0.25, 0.25, 0.1, 8) black 0x222222 at Y=0.3 evenly spaced Z=-1.6 to Z=1.6.\n"
            "ANTENNA: cylinderGeo(0.01, 0.008, 2.0, 4) dark at [1.0, 2.9, -1.5]. Whip antenna.\n"
            "HEADLIGHTS: 2x cylinderGeo(0.06, 0.06, 0.04, 6) 0xffffcc at [-0.8, 1.7, 2.28] and [0.8, 1.7, 2.28] rotation [90,0,0]."
        ),
    },
    {
        "slug": "m48-patton",
        "prompt": (
            "M48A3 Patton tank - Vietnam War US medium tank. Budget: 4000 tris.\n\n"
            "Coordinate system: front toward +Z, Y is up, tracks at Y=0. All parts connected.\n\n"
            "HULL: boxGeo(3.4, 1.0, 6.0) olive 0x556B2F at [0, 0.9, 0]. Main hull with rounded front implied by box.\n"
            "HULL FRONT SLOPE: boxGeo(3.2, 0.6, 0.06) olive at [0, 1.3, 3.0] rotation [45,0,0]. Glacis plate.\n"
            "ENGINE DECK: boxGeo(3.0, 0.06, 2.5) olive at [0, 1.42, -1.8]. Flat rear deck.\n"
            "ENGINE GRILLES: boxGeo(1.2, 0.04, 1.0) dark 0x444444 at [0.6, 1.44, -2.0]. And at [-0.6, 1.44, -2.0].\n"
            "FENDERS: Left boxGeo(0.4, 0.06, 5.5) olive at [-1.9, 1.0, 0]. Right at [1.9, 1.0, 0].\n"
            "TURRET: Use a large box with beveled feel. boxGeo(2.4, 0.8, 2.8) olive at [0, 2.0, 0.3]. Hemispherical turret.\n"
            "TURRET ROOF: boxGeo(2.2, 0.06, 2.6) olive at [0, 2.42, 0.3].\n"
            "TURRET BUSTLE: boxGeo(2.0, 0.6, 0.8) olive at [0, 2.0, -1.3]. Rear overhang.\n"
            "MANTLET: boxGeo(0.8, 0.6, 0.3) gun metal 0x3a3a3a at [0, 2.0, 1.7]. Gun mount.\n"
            "MAIN GUN: cylinderGeo(0.06, 0.05, 4.0, 8) gun metal at [0, 2.0, 3.7] rotation [90,0,0]. 90mm barrel.\n"
            "BORE EVACUATOR: cylinderGeo(0.09, 0.09, 0.2, 8) gun metal at [0, 2.0, 5.0] rotation [90,0,0]. Bulge near muzzle.\n"
            "MUZZLE BRAKE: cylinderGeo(0.07, 0.08, 0.1, 8) gun metal at [0, 2.0, 5.7] rotation [90,0,0].\n"
            "COMMANDER CUPOLA: cylinderGeo(0.35, 0.35, 0.25, 10) olive at [0.6, 2.6, 0.0]. On top right.\n"
            "CUPOLA GUN: cylinderGeo(0.02, 0.02, 0.6, 6) dark 0x2a2a2a at [0.6, 2.7, 0.5] rotation [90,0,0]. M2 .50 cal.\n"
            "LOADER HATCH: boxGeo(0.5, 0.04, 0.5) dark 0x444444 at [-0.6, 2.46, 0.5].\n"
            "SEARCHLIGHT: cylinderGeo(0.15, 0.15, 0.12, 8) 0xffffcc at [0, 2.5, 1.5] rotation [90,0,0].\n"
            "TRACKS: Left boxGeo(0.35, 0.7, 6.2) dark 0x2a2a2a at [-1.9, 0.35, 0]. Right at [1.9, 0.35, 0].\n"
            "TRACK TOP: Left boxGeo(0.35, 0.06, 5.5) dark at [-1.9, 0.72, 0]. Right mirrored.\n"
            "ROAD WHEELS: 6 per side, cylinderGeo(0.28, 0.28, 0.12, 8) 0x444444 rotation [0,0,90] at Y=0.3, evenly from Z=-2.5 to Z=2.5.\n"
            "DRIVE SPROCKET: 2x cylinderGeo(0.32, 0.32, 0.15, 8) dark at [-1.9, 0.4, -3.0] and [1.9, 0.4, -3.0] rotation [0,0,90].\n"
            "IDLER: 2x cylinderGeo(0.3, 0.3, 0.12, 8) dark at [-1.9, 0.4, 3.0] and [1.9, 0.4, 3.0] rotation [0,0,90].\n"
            "RETURN ROLLERS: 3 per side, cylinderGeo(0.1, 0.1, 0.08, 6) at Y=0.85 evenly along top."
        ),
    },
    {
        "slug": "pt76",
        "prompt": (
            "PT-76 amphibious light tank - NVA, Battle of Lang Vei. Budget: 3000 tris.\n\n"
            "Coordinate system: front toward +Z, Y is up, tracks at Y=0. All parts connected.\n\n"
            "HULL: boxGeo(3.0, 0.8, 6.0) dark green 0x3d4a2a at [0, 0.7, 0]. Low flat boat-shaped hull.\n"
            "HULL BOW: boxGeo(2.6, 0.5, 0.06) at [0, 0.9, 3.0] rotation [25,0,0]. Pointed bow plate.\n"
            "HULL TOP DECK: boxGeo(2.8, 0.04, 5.5) at [0, 1.12, 0]. Flat deck.\n"
            "HULL REAR: boxGeo(2.8, 0.7, 0.06) at [0, 0.7, -3.0]. Rear plate.\n"
            "ENGINE GRILLES: boxGeo(1.5, 0.03, 1.0) dark 0x444444 at [0, 1.14, -2.0].\n"
            "TRIM VANE: boxGeo(2.6, 0.8, 0.04) 0x4a5a3a at [0, 1.2, 3.1] rotation [45,0,0]. Raised at 45 degrees.\n"
            "TURRET: cylinderGeo(0.8, 0.8, 0.6, 12) dark green at [0, 1.5, 0.5]. Small rounded turret.\n"
            "TURRET ROOF: cylinderGeo(0.75, 0.75, 0.04, 12) at [0, 1.82, 0.5].\n"
            "COMMANDER HATCH: boxGeo(0.35, 0.03, 0.35) dark 0x444444 at [0.2, 1.86, 0.5].\n"
            "MAIN GUN: cylinderGeo(0.04, 0.035, 3.0, 8) gun metal 0x3a3a3a at [0, 1.5, 2.5] rotation [90,0,0]. 76mm barrel.\n"
            "BORE EVACUATOR: cylinderGeo(0.06, 0.06, 0.15, 8) gun metal at [0, 1.5, 3.2] rotation [90,0,0].\n"
            "COAXIAL MG: cylinderGeo(0.015, 0.015, 0.4, 4) dark at [0.15, 1.5, 1.4] rotation [90,0,0].\n"
            "TRACKS: Left boxGeo(0.3, 0.6, 6.2) dark 0x2a2a2a at [-1.65, 0.3, 0]. Right at [1.65, 0.3, 0].\n"
            "ROAD WHEELS: 6 per side, large cylinderGeo(0.25, 0.25, 0.1, 8) dark rotation [0,0,90] at Y=0.25 evenly from Z=-2.5 to Z=2.5.\n"
            "DRIVE SPROCKET: 2x cylinderGeo(0.28, 0.28, 0.12, 8) at [-1.65, 0.35, -3.0] and [1.65, 0.35, -3.0] rotation [0,0,90].\n"
            "IDLER: 2x cylinderGeo(0.25, 0.25, 0.1, 8) at [-1.65, 0.3, 3.0] and [1.65, 0.3, 3.0] rotation [0,0,90]."
        ),
    },
]

os.makedirs(OUT_DIR, exist_ok=True)

for v in vehicles:
    slug = v["slug"]
    tmp_file = f"tmp-{slug}-result.json"
    glb_file = f"{OUT_DIR}/{slug}.glb"

    print(f"\n=== Generating {slug} ===")
    payload = json.dumps({
        "prompt": v["prompt"],
        "mode": "glb",
        "category": "environment",
        "style": "low-poly",
        "includeAnimation": False,
    }).encode()

    req = urllib.request.Request(API, data=payload, headers={"Content-Type": "application/json"})
    try:
        resp = urllib.request.urlopen(req, timeout=300)
        data = json.loads(resp.read())
    except Exception as e:
        print(f"  FAILED: {e}")
        continue

    if not data.get("success"):
        print(f"  API error: {data.get('error')}")
        continue

    with open(tmp_file, "w") as f:
        json.dump(data, f)
    print(f"  Generated: {len(data.get('code', ''))} chars")

    result = subprocess.run(
        ["bun", EXPORT_SCRIPT, tmp_file, glb_file],
        capture_output=True, text=True, timeout=60
    )
    if result.returncode == 0:
        print(f"  Exported: {result.stdout.strip().split(chr(10))[-1]}")
    else:
        print(f"  Export failed: {result.stderr[:200]}")

print("\n=== Done ===")

File: scripts/gen-ground-vehicles.ts
#!/usr/bin/env bun
/**
 * Ground-vehicle GLBs — Vietnam War era.
 *
 * Regenerates the 5 existing (m113, jeep, m35, m48, pt76) with Round 3
 * helpers + adds 4 new (T-54, Ontos, ZIL-157, M42 Duster AA).
 *
 *   ANTHROPIC_API_KEY=... bun scripts/gen-ground-vehicles.ts
 */

import { mkdirSync } from 'node:fs';
import { join } from 'node:path';
import { directBatchRun } from './_direct-batch';
import { FRAME } from './_vietnam-frame';

const OUT_DIR = 'war-assets/vehicles/ground';
const AUDIT_DIR = 'war-assets/validation';
mkdirSync(OUT_DIR, { recursive: true });
mkdirSync(AUDIT_DIR, { recursive: true });

interface GroundVehicle {
  slug: string;
  prompt: string;
}

const VEHICLES: GroundVehicle[] = [
  {
    slug: 'm48-patton',
    prompt: `M48 Patton medium tank — US Army, Vietnam, olive drab. Target ~7m long, ~3000 tris.

${FRAME}

Parts:
- Low-slung hull along +X: sloped glacis plate at the front, flat engine deck at the rear, side sponsons over the tracks.
- Two track assemblies (left/right) along +X: 6 road wheels per side (cylinderZGeo), a drive sprocket at the rear, idler at the front, track lines represented as a flattened box wrapping the wheels.
- Turret: rounded squat dome (sphereGeo squashed, or a chain of cylinderZGeo + box) centered on the hull, with the gun mantlet pointing +X.
- Main gun barrel: long cylinderXGeo with a thicker mantlet section at the turret, a thermal sleeve mid-barrel, and a muzzle brake at the tip. Target barrel length ~5m.
- Commander's cupola on top of the turret with a .50 cal M2 machine gun (short cylinderXGeo) in a swivel mount.
- Exhaust panels and engine grille on the rear deck.
- Named pivot "turret" (rotates around Y for aim). Named pivot "mainGun" (child of turret, rotates around Z for elevation).

Animation clips: optional turret-sweep (slow traverse).`,
  },
  {
    slug: 'm113-apc',
    prompt: `M113 Armored Personnel Carrier — US Army, Vietnam, olive drab. Target ~5m long, aluminum-boxy. ~2500 tris.

${FRAME}

Parts:
- Aluminum box hull along +X: sloped front plate (trim vane folded down), vertical sides, flat top, open rear ramp hinged at the bottom.
- Two track assemblies (left/right): 5 road wheels per side, drive sprocket at front, idler at rear, flat box for the track runs.
- Driver's hatch on the front-left of the roof.
- Commander's cupola on top with a pintle-mounted M2 Browning .50 cal (cylinderXGeo barrel).
- Two open rear crew hatches (troop hatches).
- Rear ramp: boxGeo hinged at bottom, angled outward-down about 20 degrees to suggest "deployed".
- Radio whip antenna rising from the right rear corner — use beamBetween() from antenna base to tip (~2m tall).
- Olive-drab paint, chipped edges.

Named pivots: "cupola" (Y rotation), "m2gun" (child of cupola, X-axis for elevation), "ramp" (rear hinge).`,
  },
  {
    slug: 'pt76',
    prompt: `PT-76 amphibious light tank — NVA, Vietnam War. Dark olive. ~6.9m long, amphibious hull. ~2800 tris.

${FRAME}

Parts:
- Long boat-like amphibious hull along +X: distinctive trim vane folded at the nose, sloped front glacis, flat engine deck, squared-off stern with two water jet exhausts at the rear.
- Two track assemblies: 6 small road wheels per side, drive sprocket at rear, idler at front.
- Turret: flat conical/truncated dome centered on the hull.
- Main gun: 76mm cylinderXGeo barrel, thinner than the M48 Patton, with a bore evacuator bulge mid-barrel and a muzzle brake at the tip.
- Coax MG mount on the turret mantlet.
- Commander's cupola with one hatch.
- Two water jet exhaust pipes (cylinderXGeo) protruding from the stern.
- Antenna whip on the turret (beamBetween).

Named pivots: "turret" (Y), "mainGun" (Z for elevation).`,
  },
  {
    slug: 'm35-truck',
    prompt: `M35 "Deuce and a Half" 2.5-ton 6x6 cargo truck — US Army, Vietnam, olive drab canvas top. ~6.7m long, ~2500 tris.

${FRAME}

Parts:
- Cab along +X: blunt hood with radiator grille, two-door cab with windshield (glassMaterial), side windows, canvas-roof cab soft top.
- Front bumper with tow hooks.
- Two headlights (sphereGeo) on the grille.
- Cargo bed behind the cab: rectangular bed with side boards, tailgate at the rear.
- Canvas cover over the cargo bed (olive canvas) with a bowed top (slightly rounded using a capsuleXGeo squashed flat).
- 6 wheels: 2 front, 4 rear (dual-axle) — cylinderZGeo each, with axles spanned by beamBetween().
- Spare tire mounted behind the cab.
- Side-view mirrors (boxGeo on stalks via beamBetween).
- Fuel cans strapped to the running boards (boxGeo).
- Olive drab paint, mud-weathered.

No animation required.`,
  },
  {
    slug: 'm151-jeep',
    prompt: `M151 MUTT jeep — 1/4-ton utility truck, US Army Vietnam, olive drab. Open-top with foldable frame. ~3.4m long, ~2000 tris.

${FRAME}

Parts:
- Compact body along +X: flat hood, sloped windshield (glassMaterial) that can fold down, two-door open cab, small rear cargo area.
- Grille at the front with 7 vertical slots (thin boxGeo each).
- Two round headlights on the front fenders.
- 4 wheels (cylinderZGeo), basic rims with black tires.
- Optional M60 machine gun on a pedestal mount in the rear bed (cylinderXGeo barrel).
- Spare tire mounted on the rear.
- Canvas soft-top rolled and tied behind the rear seats (a rolled-up capsuleZGeo in olive canvas).
- Jerry can mounted on the rear bumper (boxGeo).
- Star marking on the hood (whiteMark basicMaterial disk).
- Radio antenna whip rising from the left rear fender (beamBetween, ~2m tall).

Named pivot: "pedestalGun" (if included, Y rotation).`,
  },

  // NEW additions:
  {
    slug: 't54-tank',
    prompt: `T-54 main battle tank — NVA/Soviet-pattern, dark olive. Vietnam era. ~9m long including gun, ~3200 tris.

${FRAME}

Parts:
- Low, dome-turreted hull along +X: sharply sloped glacis, flat-ish sides, slab-sided rear engine deck with vents.
- Distinctive rounded dome turret (sphereGeo squashed vertically) centered forward of hull mid-point, with the gun mantlet in a bulge at the front.
- Long 100mm main gun (cylinderXGeo) with a characteristic bore evacuator near the muzzle and a thin muzzle brake at the tip.
- Two track assemblies: 5 large road wheels per side (Soviet style — larger than M48's), drive sprocket at rear, idler at front.
- Commander's cupola on the turret with a 12.7mm DShK heavy MG (short cylinderXGeo).
- Side fuel drums (cylinderXGeo) lashed to the rear fenders.
- Snorkel tube mounted along the right side (cylinderXGeo).
- NVA or VC yellow star on the turret side.

Named pivots: "turret" (Y), "mainGun" (Z elevation).`,
  },
  {
    slug: 'ontos',
    prompt: `M50 Ontos — US Marine Corps tank destroyer, Vietnam era. Tiny tracked vehicle bristling with six 106mm recoilless rifles. ~3.8m long, ~2200 tris.

${FRAME}

Parts:
- Squat box hull along +X (aluminum armor): sloped front, flat sides, low profile (only ~2m tall to the top of the commander's hatch).
- Two small track assemblies on narrow sponsons: 3 road wheels each side.
- Small box-shaped traversing turret centered on the roof.
- Six 106mm recoilless rifle barrels (cylinderXGeo) mounted on external rails — three on each side of the turret, stacked in two rows. The tubes protrude well past the front of the vehicle.
- Spotter .50 cal machine guns (short cylinderXGeo) mounted ABOVE the 106mm barrels (one per group).
- Commander's hatch with a periscope box.
- Olive drab, USMC markings.

Named pivot: "turret" (Y rotation).`,
  },
  {
    slug: 'zil-157',
    prompt: `ZIL-157 6x6 cargo truck — NVA/Soviet-pattern supply truck moving down the Ho Chi Minh Trail. Dark green with canvas tarp over bed. ~6.8m long, ~2500 tris.

${FRAME}

Parts:
- Rounded-snout cab along +X: wide round-top grille with chrome vertical bars, two round headlights high on the fenders, bubble-style windshield (glassMaterial).
- Two-door cab with side step boards.
- Long cargo bed behind the cab with wooden side boards and a canvas cover arched over the top (create the arch with a squashed capsuleXGeo).
- 6 wheels — 2 front axle, 4 rear (dual axle) with chunky off-road treads (cylinderZGeo).
- Spare tire mounted behind the cab.
- Jerry cans (boxGeo) strapped to the running boards.
- Crates and supply bundles (a couple of boxGeo) visible poking out of the rear tarp.
- Muddy, well-used weathering.

No animation required.`,
  },
  {
    slug: 'm42-duster',
    prompt: `M42 Duster anti-aircraft vehicle — US Army Vietnam. Twin 40mm Bofors cannons on an M41 light tank chassis. Used ground-to-ground in Vietnam. ~5.8m long, ~2800 tris.

${FRAME}

Parts:
- Light tank hull along +X: sloped glacis, flat engine deck, side fenders.
- Two track assemblies: 5 road wheels each side.
- Open-top turret (a low parapet box, open to the sky) centered on the hull.
- TWIN 40mm Bofors autocannons mounted side-by-side on a common cradle inside the turret. Each barrel is a long cylinderXGeo with a flash suppressor cone at the muzzle.
- Gunner/loader positions visible inside the open turret (just named empty cubes as "gunnerSeat" / "loaderSeat" for crew placement later).
- Ammo racks (stacks of boxGeo) along the inside walls of the turret.
- Radio antenna whip on the turret rear (beamBetween, ~2.5m tall).
- Olive drab.

Named pivots: "turret" (Y rotation), "twinGuns" (Z elevation of the cannon cradle).`,
  },
];

const items = VEHICLES.map((v) => ({
  slug: v.slug,
  prompt: v.prompt,
  outPath: join(OUT_DIR, `${v.slug}.glb`),
  auditPath: join(AUDIT_DIR, `ground-${v.slug}.glb`),
}));

await directBatchRun(items, { label: 'ground', includeAnimation: true });
